cluster/credentials.go

```diff
@@ -0,0 +1,154 @@
+// This file is part of MinIO Kubernetes Cloud
+// Copyright (c) 2019 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+type UserUICredentials struct {
+	AccessKey string
+	SecretKey string
+}
+
+// createUserCredentials creates some random access/secret key pair and then stores them on k8s, if successful
+// it will create a MinIO User and attach `readwrite` policy, if successful, it will insert this credential to the
+// tenant DB
+func createUserCredentials(ctx *Context, tenantShortName string, userdID uuid.UUID) error {
+
+	userUICredentials := UserUICredentials{
+		AccessKey: RandomCharString(16),
+		SecretKey: RandomCharString(32)}
+
+	// Attempt to store in k8s, if it works, store in DB
+
+	err := storeUserUICredentialsSecret(tenantShortName, &userdID, &userUICredentials)
```

Comment:
if we fail after this part will this secret still going to be there?

Reply:
yes, until we come up with a rollback strategy for kubernetes